File: java/base/src/main/vsdk/toolkit/environment/geometry/geometricProcessing/polyhedralBoundedSolidOperators/_BoundaryRepresentationFromCurveBuildState.java
package vsdk.toolkit.environment.geometry.geometricProcessing.polyhedralBoundedSolidOperators;

import vsdk.toolkit.common.linealAlgebra.Vector3Dd;
import vsdk.toolkit.environment.geometry.volume.polyhedralBoundedSolid.PolyhedralBoundedSolid;
import vsdk.toolkit.environment.geometry.volume.polyhedralBoundedSolid.PolyhedralBoundedSolidNumericPolicy;

public class _BoundaryRepresentationFromCurveBuildState
{
    final PolyhedralBoundedSolid solid;
    boolean firstLoop;
    boolean beginningOfLoop;
    int nextVertexId;
    int lastLoopStartVertexId;
    int nextFaceId;
    Vector3Dd firstPointInLoop;
    Vector3Dd lastAcceptedPoint;
    double weldEpsilon;
    int verticesInCurrentLoop;

    public _BoundaryRepresentationFromCurveBuildState()
    {
        solid = new PolyhedralBoundedSolid();
        firstLoop = true;
        beginningOfLoop = true;
        nextVertexId = 1;
        lastLoopStartVertexId = 1;
        nextFaceId = 1;
        firstPointInLoop = new Vector3Dd();
        lastAcceptedPoint = null;
        weldEpsilon = PolyhedralBoundedSolidNumericPolicy.BREP_BIG_EPSILON;
        verticesInCurrentLoop = 0;
    }
}

File: typescript/base/src/main/vsdk/toolkit/environment/geometry/geometricProcessing/polyhedralBoundedSolidOperators/construction/_BoundaryRepresentationFromCurveBuildState.ts
import { Vector3Dd } from "../../../../../common/linealAlgebra/Vector3Dd.js";
import { PolyhedralBoundedSolid } from "../../../volume/polyhedralBoundedSolid/PolyhedralBoundedSolid.js";
import { PolyhedralBoundedSolidNumericPolicy } from "../../../volume/polyhedralBoundedSolid/PolyhedralBoundedSolidNumericPolicy.js";

export class _BoundaryRepresentationFromCurveBuildState {
    public readonly solid = new PolyhedralBoundedSolid();
    public firstLoop = true;
    public beginningOfLoop = true;
    public nextVertexId = 1;
    public lastLoopStartVertexId = 1;
    public nextFaceId = 1;
    public firstPointInLoop = new Vector3Dd();
    public lastAcceptedPoint: Vector3Dd | null = null;
    public weldEpsilon = PolyhedralBoundedSolidNumericPolicy.BREP_BIG_EPSILON;
    public verticesInCurrentLoop = 0;
}
